qiskit/result/postprocess.py

```diff
@@ -182,6 +182,13 @@ def format_statevector(vec, decimals=None):
     Returns:
         list[complex]: a list of python complex numbers.
     """
+    # pylint: disable=cyclic-import,import-outside-toplevel
+    from qiskit.quantum_info.states.statevector import Statevector
+
+    if isinstance(vec, Statevector):
+        if decimals:
+            return Statevector(np.around(vec.data, decimals=decimals))
```

Comment:
Before you had `type(vec)`, which would propagate through subclasses.  That may still be what you want to do.

Reply:
This should be fine, since there are no subclasses at the moment.